### PR TITLE
feat(ecr-assets): Support .dockerignore (faster Docker builds)

### PR DESCRIPTION
**Pull request by parisholley**
If your Dockerfile lives in the project root, CDK will copy ALL files (node_modules, etc) into the cdk out folder prior to running the `docker build` command (which is super slow). Docker has support for a `.dockerignore` file that ignores files/directories when building the context:

https://docs.docker.com/engine/reference/builder/#dockerignore-file

### CODE REVIEW

**Comment by parisholley** [comment]
@rix0rrr @eladb g2g

File: packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import assets = require('@aws-cdk/assets');
 import ecr = require('@aws-cdk/aws-ecr');
 import cdk = require('@aws-cdk/core');
-import { Token } from '@aws-cdk/core';
+import {Token} from '@aws-cdk/core';
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts]: Please revert style changes.

File: packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts [hunk 2]
```diff
@@ -75,8 +75,17 @@ export class DockerImageAsset extends cdk.Construct implements assets.IAsset {
       throw new Error(`No 'Dockerfile' found in ${dir}`);
     }
 
+    let excludes: string[] = ['.dockerignore'];
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts]: Rename to `exclude`

File: packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts [hunk 3]
```diff
@@ -112,15 +121,15 @@ export class DockerImageAsset extends cdk.Construct implements assets.IAsset {
     //
     // If adoption fails (because the repository might be twice-adopted), we
     // haven't already started using the image.
-    this.repository = new AdoptedRepository(this, 'AdoptRepository', { repositoryName });
+    this.repository = new AdoptedRepository(this, 'AdoptRepository', {repositoryName});
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts]: Revert style change

**Comment by parisholley** [comment]
@eladb done, sounds like we need some lint updates :)

**Comment by kolomied** [comment]
It seems we do not handle common "exclude everything except" pattern for ```.dockerignore``` files ([Docker docs](https://docs.docker.com/engine/reference/builder/#dockerignore-file)):

```
# Ignore everything
*
# Allow files and directories
!/src/**
```

With the current implementation the first 'star' glob effectively excludes everything from the asset folder.

**Comment by yurigorokhov** [comment]
bump, regarding not supporting pattern @kolomied brought up.

**Comment by kolomied** [comment]
@yurigorokhov I believe the problem was addressed in #4450

**Pull request merged**